adapters/p2p2consensus/proposal.go

```diff
@@ -6,69 +6,137 @@ import (
+	if err := validateBlockInfo(msg); err != nil {
+		return consensus.BlockInfo{}, err
+	}
+
+	var builder felt.Felt
+	if err := builder.SetBytesCanonical(msg.Builder.Elements); err != nil {
+		return consensus.BlockInfo{}, err
+	}
+
 	return consensus.BlockInfo{
 		BlockNumber:       msg.BlockNumber,
-		Builder:           *new(felt.Felt).SetBytes(msg.Builder.Elements),
+		Builder:           builder,
 		Timestamp:         msg.Timestamp,
 		L2GasPriceFRI:     *p2p2core.AdaptUint128(msg.L2GasPriceFri),
 		L1GasPriceWEI:     *p2p2core.AdaptUint128(msg.L1DataGasPriceWei),
 		L1DataGasPriceWEI: *p2p2core.AdaptUint128(msg.L1DataGasPriceWei),
 		EthToStrkRate:     *p2p2core.AdaptUint128(msg.EthToStrkRate),
 		L1DAMode:          core.L1DAMode(msg.L1DaMode),
-	}
+	}, nil
 }
 
-func AdaptProposalCommitment(msg *p2pconsensus.ProposalCommitment) consensus.ProposalCommitment {
+func AdaptProposalCommitment(msg *p2pconsensus.ProposalCommitment) (consensus.ProposalCommitment, error) {
+	if err := validateProposalCommitment(msg); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	var builder felt.Felt
+	if err := builder.SetBytesCanonical(msg.Builder.Elements); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	var parentCommitment felt.Felt
+	if err := parentCommitment.SetBytesCanonical(msg.ParentCommitment.Elements); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	var oldStateRoot felt.Felt
+	if err := oldStateRoot.SetBytesCanonical(msg.OldStateRoot.Elements); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	var versionConstantCommitment felt.Felt
+	if err := versionConstantCommitment.SetBytesCanonical(msg.VersionConstantCommitment.Elements); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	var stateDiffCommitment felt.Felt
+	if err := stateDiffCommitment.SetBytesCanonical(msg.StateDiffCommitment.Elements); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	var transactionCommitment felt.Felt
+	if err := transactionCommitment.SetBytesCanonical(msg.TransactionCommitment.Elements); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	var eventCommitment felt.Felt
+	if err := eventCommitment.SetBytesCanonical(msg.EventCommitment.Elements); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	var receiptCommitment felt.Felt
+	if err := receiptCommitment.SetBytesCanonical(msg.ReceiptCommitment.Elements); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	var concatenatedCounts felt.Felt
+	if err := concatenatedCounts.SetBytesCanonical(msg.ConcatenatedCounts.Elements); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	snVersion, err := semver.NewVersion(msg.ProtocolVersion)
+	if err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
 	return consensus.ProposalCommitment{
 		BlockNumber: msg.BlockNumber,
-		Builder:     *new(felt.Felt).SetBytes(msg.Builder.Elements),
+		Builder:     builder,
```

Comment:
nit: Would you mind remove the space between the fields, so we can align the field values?